watertap/property_models/anaerobic_digestion/adm1_reactions.py

```diff
@@ -0,0 +1,1734 @@
+            ("R19", "Liq", "X_I"): 0,
+        }
+
+        s_ic_rxns = ["R5", "R6", "R10", "R11", "R12"]
+
+        for R in s_ic_rxns:
+            self.rate_reaction_stoichiometry[R, "Liq", "S_IC"] = -sum(
+                self.Ci[S] * self.rate_reaction_stoichiometry[R, "Liq", S] * mw_c
+                for S in Ci_dict.keys()
+                if S != "S_IC"
+            )
+
+        for R in self.rate_reaction_idx:
+            self.rate_reaction_stoichiometry[R, "Liq", "S_cat"] = 0
+            self.rate_reaction_stoichiometry[R, "Liq", "S_an"] = 0
+
+        # Fix all the variables we just created
+        for v in self.component_objects(pyo.Var, descend_into=False):
+            v.fix()
+
+    @classmethod
+    def define_metadata(cls, obj):
+        obj.add_properties(
+            {
+                "conc_mol_co2": {"method": "_rxn_rate"},
+                "reaction_rate": {"method": "_rxn_rate"},
+                "I": {"method": "_I"},
+            }
+        )
+        obj.add_default_units(
+            {
+                "time": pyo.units.s,
+                "length": pyo.units.m,
+                "mass": pyo.units.kg,
+                "amount": pyo.units.kmol,
+                "temperature": pyo.units.K,
+            }
+        )
+
+
+class _ADM1ReactionBlock(ReactionBlockBase):
+    """
+    This Class contains methods which should be applied to Reaction Blocks as a
+    whole, rather than individual elements of indexed Reaction Blocks.
+    """
+
+    def initialize(blk, outlvl=idaeslog.NOTSET, **kwargs):
+        """
+        Initialization routine for reaction package.
+
+        Keyword Arguments:
+            outlvl : sets output level of initialization routine
+
+        Returns:
+            None
+        """
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="properties")
+        init_log.info("Initialization Complete.")
+
+
+@declare_process_block_class("ADM1ReactionBlock", block_class=_ADM1ReactionBlock)
+class ADM1ReactionBlockData(ReactionBlockDataBase):
+    """
+    ReactionBlock for ADM1.
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction
+        """
+        super().build()
+
+        # Create references to state vars
+        # Concentration
+        add_object_reference(self, "conc_mass_comp_ref", self.state_ref.conc_mass_comp)
+
+        self.rates = {
+            "R1": 1.786e-06,
+            "R2": 3.235e-06,
+            "R3": 1.187e-05,
+            "R4": 3.412e-06,
+            "R5": 3.404e-06,
+            "R6": 1.187e-05,
+            "R7": 3.185e-06,
+            "R8": 2.505e-06,
+            "R9": 3.230e-06,
+            "R10": 2.636e-06,
+            "R11": 1.220e-05,
+            "R12": 4.184e-06,
+            "R13": 9.726e-08,
+            "R14": 2.730e-07,
+            "R15": 5.626e-08,
+            "R16": 9.998e-08,
+            "R17": 3.178e-08,
+            "R18": 1.761e-07,
+            "R19": 7.338e-08,
+        }
```

Comment:
I'm not recalling where these come from exactly. I'll tag it with issue #909 to get further clarification later.

Reply:
This are values calculated with the papers outlets, this is there to initialize the reaction rates order of magnitude as they are so different with one another